.changeset/rotten-pots-itch.md

```diff
@@ -0,0 +1,59 @@
+---
+"@comet/admin": major
+---
+
+`FinalFormFileSelect` is now a simple final form wrapper around `FileSelect`
+
+Props now mirror those of `FileSelect` and are passed through to the `FileSelect` component.
+Setting `defaultProps` in the theme must now be done with `CometAdminFileSelect` instead of `CometAdminFinalFormFileSelect`.
+
+Class keys have been removed. Apply custom styling to `CometAdminFileSelect` instead of `FinalFormFileSelect`.
+
+The default value for `maxSize` has been removed.
+You may want to set the previous default value of 50 MB explicitly.
+
+```diff
+ <Field
+     name="files"
+     label="Files"
+     component={FinalFormFileSelect}
++    maxSize={50 * 1024 * 1024} // 50 MB
+ />
+```
+
+The `disableDropzone` prop has been removed.
+Use `slotProps.dropzone.hideDroppableArea` instead.
+
+```diff
+ <Field
+     name="files"
+     label="Files"
+     component={FinalFormFileSelect}
+-    disableDropzone
++    slotProps={{
++        dropzone: {
++            hideDroppableArea: true,
++        },
++    }}
+ />
+```
+
+The `disableSelectFileButton` prop has been removed.
+Use `slotProps.dropzone.hideButton` instead.
+
+```diff
+ <Field
+     name="files"
+     label="Files"
+     component={FinalFormFileSelect}
+-    disableSelectFileButton
++    slotProps={{
++        dropzone: {
++            hideButton: true,
++        },
++    }}
+ />
+```
+
+The `multiple` prop has been removed and is no longer necessary.
+The multi-file behavior is now always active unless the `maxFiles` prop is set to 1.
```

Comment:
I would keep that, Select and Autocomplete also have a multiple prop (that defaults to false)

Reply:
I don't think that makes sense. 
By default (when `maxFiles` is undefined) or when `maxFiles` is set to a value of more than one it needs to be a multi-file select anyway. If it's `1` it needs to be a single-file select. Having an additional prop to define `multiple` separately would be redundant.

Reply:
Yes it is redundant, still I would prefer that.

- in line with Select
- default value for multiple=false is more logical than maxFiles=1
- maxFiles can be optional

(But I don't care too much about that)

Reply:
I tend to agree with @nsams here. Also, I don't think the select should be multiple by default (we could achieve that with maxFiles too)

Reply:
I see your point 👍🏼 

So should the the field only support a single file by default then, instead of unlimited? 
Should the default value of `multiple` be dependent on `maxFiles`, if it's not undefined?
-> `multiple = typeof maxFiles !== "undefined" && maxFiles > 1;`

Reply:
> So should the the field only support a single file by default then, instead of unlimited?

Yes

> Should the default value of multiple be dependent on maxFiles, if it's not undefined?

IMO it would be more explicit to throw an error if maxFiles is set but multiple isn't.

Reply:
> Yes

👍🏼 

> IMO it would be more explicit to throw an error if maxFiles is set but multiple isn't.

That seems unnecessary to me... when I set `maxFiles={5}` I obviously want to have a multi file-select 🤔

Reply:
I created a PR for this: https://github.com/vivid-planet/comet/pull/2327

Reply:
> That seems unnecessary to me... when I set maxFiles={5} I obviously want to have a multi file-select

And what about `maxFiles={1}`? 😁

Reply:
> And what about maxFiles={1}? 😁

See the PR description 🙂 https://github.com/vivid-planet/comet/pull/2327